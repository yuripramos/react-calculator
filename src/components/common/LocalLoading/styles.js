import styled, { keyframes } from "styled-components";
import { white } from "../../../styles/settings";
import { rem } from "../../../styles/tools";

const loading = keyframes`
  0% {
    transform: scale(1);
  }
  20% {
    transform: scale(1, 2.2);
  }
  40% {
    transform: scale(1);
  }
`;

export const Wrapper = styled.div`
  position: relative;
  height: ${rem(40)};
`;

export const InnerWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

export const Bar = styled.div`
  display: inline-block;
  width: ${rem(3)};
  height: ${rem(18)};
  margin: 0 ${rem(2)};
  border-radius: ${rem(4)};
  animation: ${loading} 1s ease-in-out infinite;
  background-color: ${white};

  :nth-child(1) {
    animation-delay: 0;
  }

  :nth-child(2) {
    animation-delay: 0.09s;
  }

  :nth-child(3) {
    animation-delay: 0.18s;
  }

  :nth-child(4) {
    animation-delay: 0.27s;
  }
`;
