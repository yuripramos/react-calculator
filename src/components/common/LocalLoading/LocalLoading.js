import React from "react";

import { Wrapper, InnerWrapper, Bar } from "./styles";

function LocalLoading() {
  return (
    <Wrapper>
      <InnerWrapper>
        <Bar />
        <Bar />
        <Bar />
        <Bar />
      </InnerWrapper>
    </Wrapper>
  );
}

export default LocalLoading;
