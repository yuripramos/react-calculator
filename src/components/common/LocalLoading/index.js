import LocalLoading from "./LocalLoading";

export default LocalLoading;
