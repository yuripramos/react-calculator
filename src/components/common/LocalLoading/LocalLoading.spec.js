import React from "react";
import { render } from "enzyme";

import LocalLoading from "./LocalLoading";

describe("LocalLoading component", () => {
  it("should match snapshot", () => {
    expect(render(<LocalLoading />)).toMatchSnapshot();
  });
});
