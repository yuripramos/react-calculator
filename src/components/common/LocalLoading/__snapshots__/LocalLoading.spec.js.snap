// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalLoading component should match snapshot 1`] = `
<div
  class="sc-bdVaJa eyQUgi"
>
  <div
    class="sc-bwzfXH eOoZbk"
  >
    <div
      class="sc-htpNat gpdsJF"
    />
    <div
      class="sc-htpNat gpdsJF"
    />
    <div
      class="sc-htpNat gpdsJF"
    />
    <div
      class="sc-htpNat gpdsJF"
    />
  </div>
</div>
`;
